bin/check-rabbitmq-consumer-utilisation.rb

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env ruby
+
+# Check RabbitMQ consumer utilisation
+# ===
+#
+# DESCRIPTION:
+# This plugin checks the consumer utilisation percentage.
+# The fraction of time in which the queue is able to immediately deliver
+# messages to consumer. If this number drops in percentage this may result
+# in slower message delivery and indicate issues with the queue.
+#
+# Seeded from check-rabbitmq-consumers.rb
+#
+# PLATFORMS:
+#   Linux, BSD, Solaris
+#
+# DEPENDENCIES:
+#   RabbitMQ rabbitmq_management plugin
+#   gem: sensu-plugin
+#   gem: carrot-top
+#
+# LICENSE:
+# Copyright 2014 Daniel Kerwin <d.kerwin@gini.net>
+# Copyright 2014 Tim Smith <tim@cozy.co>
+# Copyright 2018 Mike Murray <37150283+monkey670@users.noreply.github.com>
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+require 'sensu-plugins-rabbitmq'
+require 'sensu-plugins-rabbitmq/check'
+
+# main plugin class
+class CheckRabbitMQConsumerUtilisation < Sensu::Plugin::RabbitMQ::Check
+  option :queue,
+         description: 'Comma separated list of RabbitMQ queues to monitor.',
+         long: '--queue queue_name',
+         proc: proc { |q| q.split(',') }
+
+  option :exclude,
+         description: 'Comma separated list of RabbitMQ queues to NOT monitor.  All others will be monitored.',
```

Comment:
Hmm I think there should maybe be an `--all` option and allow exclude to still be used with regexp to say monitor all the `app-*` queues except `app-someapp` because it needs different thresholds.

Reply:
This is a good point. We could achieve this with out the additional parameter --all. If we add another if statement into the regex section something like the below. This should give us the ability to call the code `check-rabbitmq-consumer-utilisation.rb  --regex --queue app-* --exclude app-someapp`. What do you think?

``` 
  if config[:queue] && config[:exclude]
    next unless queue['name'] =~ /#{config[:queue].first}/ && queue['name'] !~ /#{config[:exclude].first}/
  else
    next unless queue['name'] =~ /#{config[:queue].first}/
  end
```

Reply:
seems reasonable I will review this tonight in greater detail but that was more or less what I was thinking.